src/components/Home.jsx

```diff
@@ -5,29 +5,59 @@ import Upload from './Upload';
 
 const Home = ({ setActors, actors, currentScript, setCurrentScript }) => {
   const [titles, setTitles] = useState([]);
+  const [deleteError, setDeleteError] = useState('');
+  const [addError, setAddError] = useState('');
   const newActorInput = useRef('');
 
   // submit of actor name handler
   const newActorHandler = (event) => {
     event.preventDefault();
     const newActor = { name: newActorInput.current.value };
 
-    // Add the new actor to the database
+    // Add the new actor to the database, returning the new actor object
     fetch('/actors', {
       method: 'POST',
       headers: {
         'Content-Type': 'application/json',
       },
       body: JSON.stringify(newActor),
     })
-      .then(() => {
-        // set actors in state to be our new actors
-        setActors([...actors, newActor]);
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw new Error('Error when adding the actor');
+        }
+      })
+      .then((actor) => {
+        setActors([...actors, actor]);
         // reset input value to empty string to clear out old input
         newActorInput.current.value = '';
+        setAddError('');
+      })
+      .catch((error) => {
+        setAddError(error.message);
+      });
+  };
+
+  const deleteActor = (event) => {
+    const actor = event.target.value.split(',');
+    const name = actor[0];
+    const id = actor[1];
+    fetch('/actors/' + id, {
+      method: 'DELETE',
+    })
+      .then((response) => {
+        if (response.ok) {
+          // upon successful deletion, remove that actor's name from the actor list
+          setDeleteError('');
+          setActors(() => actors.filter((actor) => actor.name !== name));
```

Comment:
I don't think the filter call needs to be wrapped in an arrow function. Should be able to do just this:
```js
          setActors(actors.filter((actor) => actor.name !== name));
```